ibm/resource_ibm_network_gateway.go

```diff
@@ -0,0 +1,816 @@
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/filter"
+	"github.com/softlayer/softlayer-go/helpers/location"
+	"github.com/softlayer/softlayer-go/helpers/product"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/session"
+	"github.com/softlayer/softlayer-go/sl"
+)
+
+const packageKeyName = "NETWORK_GATEWAY_APPLIANCE"
+
+func resourceIBMNetworkGateway() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMNetworkGatewayCreate,
+		Read:     resourceIBMNetworkGatewayRead,
+		Update:   resourceIBMNetworkGatewayUpdate,
+		Delete:   resourceIBMNetworkGatewayDelete,
+		Exists:   resourceIBMNetworkGatewayExists,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
```

Comment:
We will remove `id` from all the schema as newer terraform plugin doesn't allow it to be explicity stated.
Let us remove it. You can continue using d.SetId though.

Reply:
Removed.